rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCost.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.rollingupdate.kubernetes
+
+import akka.actor.ActorSystem
+import akka.actor.ClassicActorSystemProvider
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
+import akka.actor.Props
+import akka.cluster.Cluster
+import akka.cluster.ClusterEvent
+import akka.event.Logging
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.BootstrapStep
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.Initializing
+import akka.rollingupdate.kubernetes.PodDeletionCost.Internal.NotRunning
+
+import java.util.concurrent.atomic.AtomicReference
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+final class PodDeletionCost(implicit system: ExtendedActorSystem) extends Extension {
+
+  import system.dispatcher
+
+  private val log = Logging(system, classOf[PodDeletionCost])
+
+  private val settings = KubernetesSettings(system)
+  log.debug("Settings {}", settings)
+
+  private final val startStep = new AtomicReference[BootstrapStep](NotRunning)
+
+  def start(): Unit = {
+    if (settings.podName.isEmpty) {
+      log.warning(
+        "No configuration found to extract the pod name from. " +
+        "Be sure to provide the pod name with `akka.management.pod-deletion-cost.pod-name` " +
+        "or by setting ENV variable `KUBERNETES_POD_NAME`.")
+    } else if (startStep.compareAndSet(NotRunning, Initializing)) {
+      val listener = system.systemActorOf(Props[PodDeletionCostAnnotator](), "podDeletionCostAnnotator")
+      Cluster(system).subscribe(listener, classOf[ClusterEvent.MemberUp])
+      Cluster(system).subscribe(listener, classOf[ClusterEvent.MemberRemoved])
```

Comment:
I think it's better to move these subscribe to the constructor of the actor so that it subscribes to itself `context.self`

Reply:
that would also work in case the actor is restarted

Reply:
tks, done in https://github.com/akka/akka-management/pull/1113/commits/e037463e5789c85bc091750b6e3590800a74c968